utils/theme.ts

```diff
@@ -0,0 +1,10 @@
+import {extendTheme, ThemeConfig} from '@chakra-ui/react';
+
+const config: ThemeConfig = {
+  initialColorMode: 'light',
```

Comment:
```suggestion
  initialColorMode: 'system',
```

上で指摘されている問題、この変更で解決可能だと思われます

Reply:
問題はChakraUIの挙動的に２度目のアクセスからlocalstorageの値を利用してくれるかどうかが...

Reply:
なんと、、、、！？！？！？

Reply:
ちょっと試してますのですこしお待ちを....

Reply:
今動作確認しましたが大丈夫そうですね...

実際の挙動としては
初回〜設定変更 : localstorageにデータを保存しない
設定変更 : localstorageにデータを保存
それ以降 : localstorageのデータを利用

Reply:
@Nekoya3 Fork先でビルド失敗したので確認したところTypeでsystemは定義されていないようです。

https://github.com/chakra-ui/chakra-ui/blob/75edcf41e7ff4acc2569f2169949063c164d8f6e/packages/color-mode/src/color-mode.utils.ts#L8